### PR TITLE
adding lunr.js with contextual search

### PR DESCRIPTION
**Pull request by chrisco255**
Screen Shot 2023-02-05 at 8 07 43 PM

Adds contextual search with lunr.js as local search engine.

### CODE REVIEW

File: website/src/theme/SearchBar/index.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
+import React, { useRef, useCallback, useState, useEffect } from "react";
```
Comment (DZGoldman) [inline website/src/theme/SearchBar/index.js:1]: Maybe we include an inline comment with a link to the source of this code? (I believe [this](https://code.davidawindham.com/david/til/commit/6e35b1bbe2804bb17bf91b32cc77f77303fcf7a0))

**Review by DZGoldman — CHANGES_REQUESTED** [review]

File: website/src/theme/SearchBar/index.js (hunk 1 above)
Comment (DZGoldman) [reply website/src/theme/SearchBar/index.js:1]: ...actually maybe I misunderstood - was this all just the content that was autogenned w/ swizzle? @chrisco255

Comment (chrisco255) [reply website/src/theme/SearchBar/index.js:1]: yeah it was autogenerated from Swizzle but I did add a little but of custom CSS

**Review by DZGoldman — APPROVED** [review]
lgtm!

**Pull request merged**